src/common/com/intellij/plugins/haxe/HaxeLanguageLevel.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2018 Ilya Malanin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe;
+
+import com.intellij.util.text.SemVer;
+
+public enum HaxeLanguageLevel {
+  HAXE_3(SemVer.parseFromText("3.4.0"), HaxeBundle.message("haxe.language.level.3")),
+  HAXE_4(SemVer.parseFromText("4.0.0"), HaxeBundle.message("haxe.language.level.4"));
+
+  private final SemVer version;
+  private final String presentableText;
+
+  HaxeLanguageLevel(SemVer version, String presentableText) {
+    this.version = version;
+    this.presentableText = presentableText;
+  }
+
+  public static HaxeLanguageLevel fromVersionString(String string) {
+    SemVer version = SemVer.parseFromText(string);
+    if (version != null) {
+      for (HaxeLanguageLevel level : values()) {
+        if (level.version.getMajor() == version.getMajor() && level.version.getMinor() == version.getMinor()) return level;
```

Comment:
This only works for the two specific versions you've got above.  It needs to have constants for all of the released versions since 3.1.  (Yes, people still use that.)  Or the logic needs to change to return language level based on a heuristic rather than specific version matches.

Reply:
Right now plugin mostly oriented to 3.4.7 as minimal requirement. I'm not sure if we should support any language levels for obsolete version of haxe

Reply:
Okay.

Reply:
It looks like overengeneering, I don't think that it is better way to invent feature system, when you have language features that depends on haxe version. Right now plugin mostly work with 3.4.7 language version, and if sdk will be older than that - still will be used 3.4.7 version.